common/src/main/java/com/wynntils/core/managers/ContainerQueryManager.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.managers;
+
+import com.wynntils.mc.event.ContainerSetContentEvent;
+import com.wynntils.mc.event.MenuEvent;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.model.WorldStateManager;
+import com.wynntils.wynn.utils.InventoryUtils;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import java.util.List;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.protocol.game.ServerboundContainerClickPacket;
+import net.minecraft.network.protocol.game.ServerboundContainerClosePacket;
+import net.minecraft.network.protocol.game.ServerboundSetCarriedItemPacket;
+import net.minecraft.network.protocol.game.ServerboundUseItemPacket;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.inventory.ClickType;
+import net.minecraft.world.inventory.MenuType;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ContainerQueryManager extends CoreManager {
+    private static int containerId = -2;
+    private static int transactionId = 0;
+
+    private static String lookForTitle;
+    private static final String QUESTS_TITLE = "§0[Pg. 1] §8mag_icus'§0 Quests";
+    private static final String QUESTS_TITLE_2 = "§0[Pg. 2] §8mag_icus'§0 Quests";
+    private static final String QUESTS_TITLE_3 = "§0[Pg. 3] §8mag_icus'§0 Quests";
```

Comment:
Is this intentional?

Reply:
Yeah, very much draft. :-) I just wanted to show a bit what I'm doing and kind of where I'm heading.

The concrete matching stuff needs to be moved to a QuestBookExtractionModel or something like that.